cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/apply/ApplicationRequestsPlaner.java

```diff
@@ -64,62 +69,83 @@ private void addCreateAppRequest(ApplicationBean affectedObject) throws Creation
+     * @throws ApplyException           if an error during the apply logic occurs.
+     *                                  May contain another exception inside with
+     *                                  more details
+     * @throws IllegalArgumentException if the newObject is neither an
+     *                                  ApplicationBean or an
+     *                                  ApplicationManifestBean
      */
     @Override
     public void visitRemovedObject(CfRemovedObject removedObject) {
-        //TODO: later US
+              
+        Object affectedObject = removedObject.getAffectedObject();
+        if (affectedObject instanceof ApplicationBean) {
+            addRemoveAppRequest();
+        } else if (!(affectedObject instanceof ApplicationManifestBean)) {
+            throw new IllegalArgumentException("Only changes of applications and manifests are permitted.");
+        }
         return;
     }
 
+    private void addRemoveAppRequest() {
+        this.requests.add(this.appOperations.remove(this.applicationName));
+    }
+
     /**
      * Creates the requests CfContainerChange
+     * 
      * @param containerChange the CfContainerChange to be visited
      */
     @Override
     public void visitContainerChange(CfContainerChange containerChange) {
-        //TODO: later US
+        // TODO: later US
         return;
     }
 
     /**
      * Creates the requests for CfMapChange
+     * 
      * @param mapChange the CfMapChange to be visited
      */
     @Override
     public void visitMapChange(CfMapChange mapChange) {
-        //TODO: later US
+        // TODO: later US
         return;
     }
 
     /**
-     * Creates the requests for one application.
-     * @param appOperations the ApplicationOperations object used for
-     * @param applicationName the name of the application
-     * @param applicationChanges a list with all the Changes found during diff for that specific application
-     * @throws ApplyException if an error during the apply logic occurs. May contain another exception inside
-     * with more details
-     * @throws IllegalArgumentException if the newObject is neither an ApplicationBean or an ApplicationManifestBean
-     * @return Flux of all requests that are required to apply the changes
+     * Creates the remove/create requests for one application.
+     * 
+     * @param appOperations      the ApplicationOperations object used for
+     * @param applicationName    the name of the application
+     * @param applicationChanges a list with all the Changes found during diff for
+     *                           that specific application
+     * @return flux of all requests that are required to apply the changes
      */
-    public static Flux<Void> create(ApplicationsOperations appOperations, String applicationName,
-                                    List<CfChange> applicationChanges) {
+    public static Flux<Void> apply(ApplicationsOperations appOperations, String applicationName,
```

Comment:
I would give another name then just apply. 
Apply sounds generic for me. 

In fact, you just get the requests, so maybe createApplicationRequest or something like that.